### PR TITLE
feat: check In dialog in QR scan activity

### LINKED ISSUE #230: Show attendee check in dialog in QR scan activity
To be done after #224 is merged

Currently QR scan activity is started for results from attendees fragment and on successful scan attendee check in dialog is shown in attendees fragment.

Expected: Attendee check in dialog (from #224) shall be shown in QR scan activity itself.

### PR DESCRIPTION
**Pull request by mejariamol**
- modifies qr scan presenter's behaviour on barcode detection such that data never emits detected value if dialog is shown already.

fixes #230

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal please review. Will add tests once approved.

File: app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java [hunk 1]
```diff
@@ -251,4 +246,14 @@ public void startScan() {
...
     }
+
+    public void showToggleDialog(long attendeeId) {
+        BottomSheetDialogFragment bottomSheetDialogFragment = AttendeeCheckInFragment.newInstance(attendeeId);
+        bottomSheetDialogFragment.show(getSupportFragmentManager(), bottomSheetDialogFragment.getTag());
+    }
+
+    @Override
+    public void onDialogDismissed() {
+        presenter.allowScan(true);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: This is not correct way, instead, attach a dismiss listener to bottom fragment and simply stop scanning when it is shown and resume when dismissed

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: >simply stop scanning when it is shown and resume when dismissed

couldn't figure out how to stop and resume scanning. will need your help. @iamareebjamal 
This is just work around :smile:

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: There are two ways, one is like yours, just simpler, use a boolean as a filter and just change its value in the presenter using a method, it'll automatically stop and resume. But there is no need to apply conditionals and all. Just change boolean to false when dialog is shown and to true when it is hidden

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: >Just change boolean to false when dialog is shown and to true when it is hidden

if we wait till dialog is shown to set boolean false, it let presenter to create more than one dialogs. Hence the boolean is set to false in filter itself and changed to true if attendee is not found, hence all that conditions. You may take a look again.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: I meant as soon as presenter triggers the dialog to be shown -> onScannedAttendee, you automatically set to false, then it won't create this problem

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: but till it comes to`scanQRView.onScannedAttendee(attendee);` , `data` emits more values and hence dialogs will be created.

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: in meanwhile when attendee is searched in the list

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: OK, let me see

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: @iamareebjamal did you take a look into this?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: Yes, currently testing my implementation

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: I am sending the PR but it is showing 20 commits in between and 40 files changed

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: Sent PR, please check

File: app/src/main/java/org/fossasia/openevent/app/event/checkin/AttendeeCheckInFragment.java [hunk 2]
```diff
@@ -67,9 +71,27 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     @Override
     public void onDetach() {
         super.onDetach();
+        if (getActivity() instanceof OnDismissListener) {
+            ((OnDismissListener)getActivity()).onDialogDismissed();
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/checkin/AttendeeCheckInFragment.java]: Please remove this

File: app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java [hunk 3]
```diff
@@ -48,7 +47,7 @@
 
 import static org.fossasia.openevent.app.utils.ViewUtils.showView;
 
-public class ScanQRActivity extends AppCompatActivity implements IScanQRView {
+public class ScanQRActivity extends AppCompatActivity implements IScanQRView, AttendeeCheckInFragment.OnDismissListener {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java]: Please remove this implements, 

Ummm. Please fix the changes. There was some problem in the merging I guess

File: app/src/test/java/org/fossasia/openevent/app/presenter/ScanQRPresenterTest.java [hunk 4]
```diff
@@ -293,30 +293,6 @@ public void shouldSendSameBarcodeOnlyOnce() {
...
-        InOrder inOrder = inOrder(scanQRView);
-
-        inOrder.verify(scanQRView).showBarcodeData(barcode1.displayValue);
-        inOrder.verify(scanQRView).showBarcodeData(barcode2.displayValue);
-        inOrder.verify(scanQRView).showBarcodeData(barcode1.displayValue);
-        inOrder.verify(scanQRView).showBarcodeData(barcode2.displayValue);
-        inOrder.verify(scanQRView).showBarcodeData(barcode1.displayValue);
-        inOrder.verifyNoMoreInteractions();
-    }
-
```
Comment (iamareebjamal) [inline app/src/test/java/org/fossasia/openevent/app/presenter/ScanQRPresenterTest.java]: Readd these tests please

**Comment by mejariamol** [comment]
@Shailesh351 @arp95 please review

**Pull request merged**